### PR TITLE
Restoring alpha/beta/rc version semantics

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Restoring alpha/beta/rc version semantics
 
### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/622
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@tlfeng mind please taking a look? Addressed the TODO item and restored apha/beta/rc version semantics, if it makes sense, thank you.

**Comment by nknize** [comment]
What's the reasoning for adding build qualifiers back to `Version.*`? It was moved to a build flag (`build.version_qualifier`) to begin with because it made (index, transport, api, etc) bwc support logic very clumsy to begin with.

**Comment by reta** [comment]
Thanks a lot for the feedback @nknize, the Version API was migrated only partially (see please https://github.com/opensearch-project/OpenSearch/pull/1112/files#diff-5fe8bce6f41cbba30ab4dad09445194a255d5dc8a8fecb94e0152351b84b1588L179):
 - the plugin compatibility issue (see please  https://github.com/opensearch-project/OpenSearch/issues/622), the `beta` qualifier was breaking the version parsing  
 - the Version API does actually expose the isBeta() / isAlpha() / isRC() but it does not properly encode/decode it (since it was cut to 3 components, the build qualifier was dropped)

The `build.version_qualifier` is totally fine, however the final version is built out of `opensearch`-`build.version_qualifier` (1.0.0-beta1 fe) was unparseable from the server perspective (the Version toXxx/fromXxx).

Hope it make sense, thank you!

**Comment by nknize** [comment]
:+1:  good deal! Thanks for the explanation. I'll dig in shortly if @tlfeng can't get around to it. Quickly skimming the changes look good to me. The string representation should absolutely be consistent with the id number; and I like that it doesn't bring back the constants, enabling us to bump the id number across Alpha|Beta|RC releases without changing compatibility logic.

**Comment by reta** [comment]
:+1:  Thank you, @nknize , please let me know if you would like me to add more documentation or alike, from testing perspective I think the coverage is good.

**Comment by nknize** [comment]
start gradle check

**Comment by reta** [comment]
@dblock pushing my luck here :D but do you have an opinion on this one? Unfortunately @nknize seems to be unavailable at the moment, thank you!

File: server/src/main/java/org/opensearch/Version.java [hunk 1]
```diff
@@ -381,18 +397,10 @@ public String toString() {
...
             sb.append(build);
         } else if (isBeta()) {
-            if (major >= 2) {
-                sb.append("-beta");
-            } else {
-                sb.append(".Beta");
-            }
-            sb.append(major < 5 ? build : build-25);
+            sb.append("-beta");
+            sb.append(build-25);
```
Comment (dblock) [inline server/src/main/java/org/opensearch/Version.java:401]: nit: spaces around `-`

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
This LGTM. I would wait for @tlfeng or @nknize for a bit longer to take a look though, this whole versioning business is wild to me.

File: server/src/main/java/org/opensearch/Version.java (hunk 1 above)
Comment (reta) [reply server/src/main/java/org/opensearch/Version.java]: Fixed, thank you!

**Comment by dblock** [comment]
start gradle check

**Comment by reta** [comment]
Had to merge with master, the version was changed (1.0.0 -> 1.0.0)

**Review by nknize — APPROVED** [review]
This LGTM.. once gradle check passes I'm good for this to be merged.

**Comment by reta** [comment]
Thank you @nknize and @tlfeng , checks are green :-)

**Comment by nknize** [comment]
Thanks @reta  can you open a backport PR for 1.x branch?

**Comment by reta** [comment]
Yes, will do tomorrow morning, thank you @nknize !

**Comment by nknize** [comment]
No problem @reta.  Thank you for the contribution! I'll keep an eye out for the backport.

**Pull request merged**